src/main/python/tests/algorithms/test_pca.py

```diff
@@ -48,10 +51,36 @@ def test_500x2(self):
         m1 = self.generate_matrices_for_pca(30, seed=1304)
         X = self.sds.from_numpy( m1)
         # print(features)
-        [res, model, _, _ ] = pca(X, K=1, scale="FALSE", center="FALSE").compute()
+        [res, model, _, _] = pca(X, K=1, scale="FALSE", center="FALSE").compute()
         for (x, y) in zip(m1, res):
             self.assertTrue((x[0] > 0 and y > 0) or (x[0] < 0 and y < 0))
 
+    def test_500x2b(self):
+        """
+        This test constructs a line of values in 2d space.
+        That if fit correctly maps perfectly to 1d space.
+        The check is simply if the input value was positive
+        then the output value should be similar.
+        """
+        m1 = self.generate_matrices_for_pca(30, seed=1304)
+        node0 = self.sds.from_numpy(m1)
+        # print(features)
+        node1 = List(node0.sds_context, 'pca', named_input_nodes={"X": node0, "K": 1, "scale": "FALSE", "center": "FALSE"},
+                     outputs=[("res", OutputType.MATRIX), ("model", OutputType.MATRIX), ("scale", OutputType.MATRIX), ("center", OutputType.MATRIX)])
+        node2 = node1["res"].abs()
+        result = node2.compute(verbose=True)
+
+    def test_multiple_outputs(self):
+        # Added a second test function because test_500x2b doesn't account for the case where multiple outputs of a node which provides
+        # multiple outputs are used
+        node0 = self.sds.from_numpy(np.array([1, 2, 3, 4, 5, 6, 7, 8, 9]))
+        node1 = self.sds.from_numpy(np.array([10, 20, 30, 40, 50, 60, 70, 80, 90]))
+        params_dict = {'X': node0, 'Y': node1}
+        node2 = List(self.sds, 'split', named_input_nodes=params_dict,
+                     outputs=[("X_train", OutputType.MATRIX), ("X_test", OutputType.MATRIX), ("Y_train", OutputType.MATRIX), ("Y_test", OutputType.MATRIX)])
+        node3 = node2["X_train"] + node2["Y_train"]
```

Comment:
this probably does not work, because the dimensions of X_train and Y_train is not equal.

I would like to have a new test file only for lists. located at 
src/main/python/tests/list/...

Again i would really like having a dedicated tests that looks like this:

```Python
m1 = self.sds.from_numpy(...)
m2 = self.sds.from_numpy(...)
list = self.sds.list(m1,m2)
res = list[1].compute()
self.assertTrue(np.allclose(m2, res))
```
The important thing is that it also verify the output


The only missing feature to make this work is the "sds.list(...)" function.
it would look something like
```python 
def list(self, *args: Sequence[VALID_INPUT_TYPES, ** kwargs: Dict[str, VALID_INPUT_TYPES]) -> 'Operation_Node':
    ...
    if kwargs not None and args not None:
        raise SOME ERROR...
    elif kwargs not None:
        return List(self, 'list', named_input_nodes=kwargs, outputs = kwargs) 
    else if args not None:
        return List(self, 'list', unnamed_input_nodes= args, outputs = args)

```

Reply:
and this:

```python 
m1 = self.sds.from_numpy(...)
m2 = self.sds.from_numpy(...)
list = self.sds.list(m1,m2)
tmp = list[1] + 2
res = tmp.compute()
assertTrue(np.allclose(m2 + 2, res))
```